src/withStyles.jsx

```diff
@@ -67,7 +70,23 @@ export function withStyles(
             }}
           />
         );
-      }
+      };
+    } else {
+      // WrappedComponent is a function, so we can avoid a little overhead by
+      // calling it directly instead of rendering it. This avoids React
+      // lifecycle overhead for the wrapped component and may not be necessary
+      // in a future version of React.
+      WithStyles.prototype.render = function render() {
+        const { themeName } = this.context;
+        flushIfNecessary();
+        return WrappedComponent({
+          ...this.props,
+          ...{
```

Comment:
interesting enough, babel will actually compile this into the same thing :) I agree though that it adds unnecessary complexity making it harder to read